openedx/frontend-app-discussions
# feat: add support for question type posts [BD-38] [TNL-8723] [BB-4884]

Question type posts/thread need special handling since they deal with endorsed and un-endorsed comments separately. We need to track pagination for endorsed and unendorsed comments separately, and display endorsed comments above all others. 


**JIRA tickets**: 
- [TNL-8723](https://openedx.atlassian.net/browse/TNL-8723)

**Dependencies**: 
- #19 makes a lot of UI changes to the affected pages here
- #20  adds support for pagination and this touches pagination a lot

So the above two PRs should be merged before this PR can be finalised.

**Screenshots**: 

https://user-images.githubusercontent.com/118837/135241028-5b60a92a-e6e5-4a20-b543-ceb67fc2e68a.mp4

This PR includes UI changes to align the UI closer to the newer designs. However, the aim currently isn't to reach a 100% match, but mainly to make incremental changes that mostly change the pages that were already touched. 

**Testing instructions**:

1. Create a question type post
2. Add endorsed and unendorsed replies to it using the existing forum interface at `/courses/{course_id}/discussion/forum/`
2. Make sure that both endorsed and un-endorsed comments are showing up 

**Reviewers**
